lambda-api-gateway-proxy-http4s/src/main/scala/feral/lambda/ApiGatewayProxyHttp4sLambda.scala

```diff
@@ -47,7 +48,8 @@ abstract class ApiGatewayProxyHttp4sLambda
   override final def apply(
       event: ApiGatewayProxyEventV2,
       context: Context,
-      routes: HttpRoutes[IO]): IO[Some[ApiGatewayProxyStructuredResultV2]] =
+      routes: HttpRoutes[IO])(
+      implicit T: Trace[IO]): IO[Some[ApiGatewayProxyStructuredResultV2]] =
```

Comment:
How does this `Trace` propagate to userland code? Apologies if I'm missing something.

Reply:
No, good catch. I've only made changes to `IOLambda` and the `IOLambdaPlatform`s to make sure it worked at that level. We'd need to add the implicit `Trace` to the other abstract methods if we want to do this consistently.

Reply:
Aha, I thought it might be using some `IOLocal` magic :)

Reply:
I need to think a little more on how the `ApiGatewayProxyHttp4sLambda` fits in here. (Open to suggestions if you see a way to do it with the current structure.)

We may need to change `routes` to be `def routes: Resource[IO, HttpRoutes[Kleisli[IO, Span[IO], *]]]`? But even that doesn't fit perfectly if all we have here is the implicit `Trace[IO]`. Not sure yet.

Reply:
There's this https://github.com/tpolecat/natchez-http4s

Reply:
I kind of punted and added the Natchez server middleware to `ApiGatewayProxyHttp4sLambda`. The main problem I see is that doesn't make the `Trace[IO]` capability available to anything defined in `def routes: Resource[IO, HttpRoutes[IO]]`, meaning I'm not sure how application code could open its own spans or propagate the trace onwards if it makes calls to other services.

@armanbilge it looked like you started a new branch that (from the commit messages) might be reworking the structure a little bit to help with this. I'm going to pause here to see what you come up with. 🙂